### DISCUSSION TITLE
How to import types from valtio?

### DISCUSSION THREAD

**Discussion by teetotum** [pmndrs/valtio#971] [Q&A] [answered]
The import of `type Op` is not recognized: `Module '"valtio"' has no exported member 'Op'.ts(2305)`
What would be the correct import for valtio types?
I'm on valtio 1.13.0

```ts
import { subscribe, proxy, type Op } from 'valtio'

// ...
// const state = ...
// ...

subscribe(proxy(state), (ops: Op[]) => {
  // ...
})
```

**Comment by overthemike** [comment, marked as answer]
valtio doesn't export the type you're looking for. If you want to make your own:
```ts
type Path = (string | symbol)[]
type Op =
  | [op: 'set', path: Path, value: unknown, prevValue: unknown]
  | [op: 'delete', path: Path, prevValue: unknown]
```
Created an issue to have this looked into: #972

**Comment by dai-shi** [comment]
Are you using it? Does this work for you?

```js
subscribe(proxy(state), () => {
  // ...
})```

**Reply by teetotum** [reply]
I figured out that I wasn't really using the `op` argument in my callback as I was really only accessing the `state` object whose reference was captured inside the callback closure anyway.
So: No, I'm not using it and the `Op` type can stay internal as far as I'm concerned.

**Reply by dai-shi** [reply]
👍 Thanks for confirming.
